채널의 슬라이드를 전체화면으로 볼 수 있다.

happydhKim opened the pull request:
# 채널의 슬라이드를 전체화면으로 볼 수 있다.

## 해당 이슈 📎

- [Epic#27, Epic#34] 채널의 슬라이드를 전체화면으로 볼 수 있다.(#85)

## 변경 사항 🛠

- 슬라이드를 전체화면으로 볼 수 있다.
- 화면 동기화도 정상적으로 작동한다.

## 테스트 ✨

>없음

## 리뷰어 참고 사항 🙋‍♀️

> 없음

happydhKim committed:
[ FEATURE ] 슬라이드 전체화면을 위해 모듈 설치 (#85)

Why
- 직접 전체화면 모듈을 구현하려고 했으나, 일정이 밀리는 이유로 모듈로 임시 대체.
- 추후에 시간이 남는다면 모듈을 직접 구현.

What
- react-full-screen 모듈 설치
- https://www.npmjs.com/package/react-full-screen 에서 확인 가능.

happydhKim committed:
[ FEATURE ] 슬라이드 전체화면 및 전체화면 동기화 (#85)

Why
- 전체화면 버튼을 누르면 슬라이드가 전체화면이 되야 한다.

What
- react-full-screen 모듈을 이용해 전체화면 구현.
- 액션을 취하면, 전체화면이 종료되게 구현.

happydhKim committed:
[ REFACTOR ] 슬라이드 전체화면을 종료할 때, 상태가 안변하던 현상 수정 (#85)

Why
- 전체화면을 종료하면 상태전환이 안됨.

What
- 상태를 onChange로 받아 전체화면과 아닌 상태를 구분시킴.

always-awake commented on web/src/components/channel/Slide/SlideViewer/SlideViewer.jsx:
```diff
@@ -60,7 +60,7 @@ const SlideViewer = (props) => {
         <FullScreen
           enabled={isFullScreen}
-          onKeyDown={setFullScreen(false)}
+          onChange={(e) => setFullScreen(e)}
```
event 객체를 바로 setFullScreen의 param으로 넘기는 것 같은데 잘 이해가 안가는 부분이네요! 🤷‍♀️

happydhKim replied:
🦉음 설명이 필요할까요? 어떻게 생각하세요?

happydhKim replied:
수정했습니다.!

always-awake commented on web/package.json:
```diff
@@ -27,6 +27,7 @@
     "react": "^16.11.0",
     "react-apollo": "^3.1.3",
     "react-dom": "^16.11.0",
+    "react-full-screen": "^0.2.4",
```
적절한 모듈을 찾으신 것 같네용 👍

happydhKim committed:
[ REFACTOR ] 리뷰에서 받은 내용을 수정 (#85)

Why
- 리뷰에서 받은 내용을 수정해야 함.

What
- 전체화면의 상태를 변경해주는 부분을 가독성 있게 수정.
- screenChange 함수를 추가.

happydhKim committed:
Merge branch 'develop' into feature/issue85
